portfolio/src/main/webapp/index.html

```diff
@@ -1,49 +1,48 @@
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css" integrity="sha384-9aIt2nRpC12Uk9gS9baDl411NQApFmC26EwAOH8WgZl5MYYxFfc+NcPb1dKGj7Sk" crossorigin="anonymous">
+    <link rel="stylesheet" href="style.css">
+    <script src="script.js"></script>
+    </head>
+  <body>
+    <!-- Navigation Bar -->
+    <nav class="navbar navbar-expand-lg navbar-dark fixed-top">
+      <div class="container">
+        <a class="navbar-brand" href="#">Home</a>
 
-                <!-- Hamburger Menu on small resolutions -->
-                <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation">
-                    <span class="navbar-toggler-icon"></span>
-                </button>
+        <!-- Hamburger Menu on small resolutions -->
+        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation">
+          <span class="navbar-toggler-icon"></span>
+        </button>
 
-                <!-- Left side of Navigation Bar -->
-                <div class="collapse navbar-collapse" id="navbarResponsive">
-                    <ul class="navbar-nav ml-auto">
-                        <li class="nav-item"><a class="nav-link" href="#">About</a></li>
-                        <li class="nav-item"><a class="nav-link" href="#">Work</a></li>
-                        <li class="nav-item"><a class="nav-link" href="#">Passions</a></li>
-                        <li class="nav-item"><a class="nav-link" href="#">Contact</a>
-                        </li>
-                    </ul>
-                </div>
-            </div>
-        </nav>
+        <!-- Left side of Navigation Bar -->
+        <div class="collapse navbar-collapse" id="navbarResponsive">
+          <ul class="navbar-nav ml-auto">
+            <li class="nav-item"><a class="nav-link" href="#">About</a></li>
+            <li class="nav-item"><a class="nav-link" href="#">Work</a></li>
+            <li class="nav-item"><a class="nav-link" href="#">Passions</a></li>
+            <li class="nav-item"><a class="nav-link" href="#">Contact</a></li>
+          </ul>
+        </div>
+      </div>
+    </nav>
         
-        <!-- Masthead section -->
-        <header class="masthead">
-            <div class="container h-100">
-                <div class="row h-100 align-items-center">
-                <div class="col-12 text-center">
-                    <h1 class="font-weight-light">Welcome!</h1>
-                    <p class="lead">Andrei Topala Portofolio</p>
-                </div>
-                </div>
-            </div>
-        </header>
-    </body>
+    <!-- Masthead section -->
+    <header class="masthead">
+      <div class="container h-100">
+        <div class="row h-100 align-items-center">
```

Comment:
No need to change this here: but here's what I found useful about naming CSS classes and id selectors.

I think it's beneficial to give CSS classes names like "masthead" or "nav-item" or "nav-link" or "lead" that are semantically meaningful, and do not explain what style the CSS class applies to the HTML elements. But avoid names such as "textfont-weight-light" (maybe instead "welcome-banner" or just "welcome").

Why? Naming the CSS classes based on their purpose on the HTML page allows you to separate content and styling better. If you want to have all the "nav-item" elements be centered / bold / different color, you only need to modify our CSS file.

If you want to change your welcome banner to be in bold when using class "font-weight-light", you would have to change both the HTML and the CSS file.  Also `class="font-weight-light"` is not as different as embedded CSS `style=font-weight: ...`, which I usually try to avoid putting into the HTML.

Let me know if you have further questions on this.

Reply:
The CSS classes that seem to be very specific and less generic come from the Bootstrap framework https://getbootstrap.com. For example, you can find that <code> class="font-weight-light" </code> has the following meaning https://bootstrapshuffle.com/classes/text/font-weight-light.
Of course, I get your point. The only class that I built from scratch was "masthead", which has meaningful and generic name.

Reply:
Oh interesting, I did not know about this framework, so put that on my reading list to have a look at, thank you.

For context, I mostly wanted to make sure you're aware of the idea behind separation of content and presentation because it's a useful concept. 

Even with the bootstrap framework, some of my point probably remains valid, that these classes are convenient shorthands for certain style changes. But we can discuss this another day.